File: hw/chalan/u2/hw19Inheritance/Farmsystem Python/model/FarmAnimal.py
class FarmAnimal:
    def __init__(self, id, breed, born_on, gender, is_able_to_reproduce, weight, cage):
        self.id = id
        self.breed = breed
        self.born_on = born_on
        self.gender = gender
        self.is_able_to_reproduce = is_able_to_reproduce
        self.weight = weight
        self.cage = cage

    def get_age_in_months(self):
        return 0 

    def __str__(self):
        return f"FarmAnimal{{id={self.id}, breed={self.breed}, born_on={self.born_on}, gender={self.gender}, weight={self.weight}, cage={self.cage}}}"

File: hw/rojas/u2/hw19Inheritance/hw19InheritancePythonAndJavaScript/hw20JavaScript/ec.edu.espe.model/FarmAnimal.js
import { Cage } from './Cage.js';

export class FarmAnimal {
    constructor(id, breed, bornOn, gender, isAbleToReproduce, weight, cage) {
        this.id = id;
        this.breed = breed;
        this.bornOn = bornOn; // Debe ser un objeto Date de JS
        this.gender = gender;
        this.isAbleToReproduce = isAbleToReproduce;
        this.weight = weight;
        this.cage = cage; // Espera un objeto de tipo Cage
    }

    getAgeInMonths() {
        const today = new Date();
        // bornOn debe ser un objeto Date. Si viene como string, asegúrate de convertirlo.
        let months = (today.getFullYear() - this.bornOn.getFullYear()) * 12;
        months -= this.bornOn.getMonth();
        months += today.getMonth();
        
        // Ajuste si el día actual es menor al día de nacimiento
        if (today.getDate() < this.bornOn.getDate()) {
            months--;
        }
        
        return months <= 0 ? 0 : months;
    }
}